src/devices/Mpu9250/Mpu9250.cs

```diff
@@ -0,0 +1,983 @@
+            accFactoryTrim.Z = (float)((2620 / 1 << gyroAccScale) * (Math.Pow(1.01, accSelfTest.Z - 1.0)));
+            gyroFactoryTrim.X = (float)((2620 / 1 << gyroAccScale) * (Math.Pow(1.01, gyroSelfTest.X - 1.0)));
+            gyroFactoryTrim.Y = (float)((2620 / 1 << gyroAccScale) * (Math.Pow(1.01, gyroSelfTest.Y - 1.0)));
+            gyroFactoryTrim.Z = (float)((2620 / 1 << gyroAccScale) * (Math.Pow(1.01, gyroSelfTest.Z - 1.0)));
+
+            if (gyroFactoryTrim.X != 0)
+            {
+                gyroAvegage.X = (gyroSelfTestAverage.X - gyroAvegage.X) / gyroFactoryTrim.X;
+            }
+            else
+            {
+                gyroAvegage.X = Math.Abs(gyroSelfTestAverage.X - gyroAvegage.X);
+            }
+
+            if (gyroFactoryTrim.Y != 0)
+            {
+                gyroAvegage.Y = (gyroSelfTestAverage.Y - gyroAvegage.Y) / gyroFactoryTrim.Y;
+            }
+            else
+            {
+                gyroAvegage.Y = Math.Abs(gyroSelfTestAverage.Y - gyroAvegage.Y);
+            }
+
+            if (gyroFactoryTrim.Z != 0)
+            {
+                gyroAvegage.Z = (gyroSelfTestAverage.Z - gyroAvegage.Z) / gyroFactoryTrim.Z;
+            }
+            else
+            {
+                gyroAvegage.Z = Math.Abs(gyroSelfTestAverage.Z - gyroAvegage.Z);
+            }
+
+            // Accelerator
+            if (accFactoryTrim.X != 0)
+            {
+                accAverage.X = (accSelfTestAverage.X - accAverage.X) / accFactoryTrim.X;
+            }
+            else
+            {
+                accAverage.X = Math.Abs(accSelfTestAverage.X - accSelfTestAverage.X);
+            }
+
+            if (accFactoryTrim.Y != 0)
+            {
+                accAverage.Y = (accSelfTestAverage.Y - accAverage.Y) / accFactoryTrim.Y;
+            }
+            else
+            {
+                accAverage.Y = Math.Abs(accSelfTestAverage.Y - accSelfTestAverage.Y);
+            }
+
+            if (accFactoryTrim.Z != 0)
+            {
+                accAverage.Z = (accSelfTestAverage.Z - accAverage.Z) / accFactoryTrim.Z;
+            }
+            else
+            {
+                accAverage.Z = Math.Abs(accSelfTestAverage.Z - accSelfTestAverage.Z);
+            }
+
+            return new Tuple<Vector3, Vector3>(gyroAvegage, accAverage);
+        }
+
+        #endregion
+
+        #region I2C
+
+        /// <summary>
+        /// Write data on any of the I2C slave attached to the MPU9250
+        /// </summary>
+        /// <param name="i2cChannel">The slave channel to attached to the I2C device</param>
+        /// <param name="address">The I2C address of the slave I2C element</param>
+        /// <param name="register">The register to write to the slave I2C element</param>
+        /// <param name="data">The byte data to write to the slave I2C element</param>
+        public void I2cWrite(I2cChannel i2cChannel, byte address, byte register, byte data)
```

Comment:
What are the use cases of these two methods? (I2c read and write) In your sample it doesn't seem like you are showing how these would be used so it is not clear to me why you would need this. Is this device also working as a sort of i2c extender?

Reply:
if so, I think that the name of this API is confusing, and we should probably instead rename it to WriteByteToSlave and ReadByteFromSlave

Reply:
Correct, it's about writing and reading to and from slave, renaming